app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt

```diff
@@ -137,13 +71,11 @@ class SearchFragment : Fragment() {
         searchView = menu.findItem(R.id.search_item).actionView as SearchView
         val queryListener = object : SearchView.OnQueryTextListener {
             override fun onQueryTextSubmit(query: String): Boolean {
-                // Do your search
-                searchViewModel.searchEvent = query
-                if (searchViewModel.savedLocation != null && TextUtils.isEmpty(rootView.locationTextView.text.toString()) && rootView.timeTextView.text == "Anytime")
-                    searchViewModel.loadEvents(searchViewModel.savedLocation.nullToEmpty(), searchViewModel.savedDate.nullToEmpty())
-                else
-                    searchViewModel.loadEvents(rootView.locationTextView.text.toString().nullToEmpty(), rootView.timeTextView.text.toString().nullToEmpty())
-                loadEventsAgain = true
+                val intent = Intent(activity, SearchActivity::class.java)
+                intent.putExtra("query", query)
+                intent.putExtra("location", rootView.locationTextView.text.toString().nullToEmpty())
+                intent.putExtra("date", rootView.timeTextView.text.toString().nullToEmpty())
```

Comment:
Why don't you declare `date` `location` ... etc as constants

Reply:
Got it! Sorry for that.